src/formatDate.js

```diff
@@ -7,8 +7,70 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDateArr = [];
+  let month = '';
+  let day = '';
+  let year = '';
+
+  const separator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateArr = date.split(separator);
+
+  const requiredYearFormat = toFormat.find(
+    (element) => element === 'YYYY' || element === 'YY',
+  );
+
+  const changeYearFormat = () => {
+    if (requiredYearFormat.length === 4 && year.length === 4) {
+      return year;
+    }
+
+    if (requiredYearFormat.length === 4 && year < 30) {
```

Comment:
There is a potential issue in the logic for converting the year format. You are comparing 'year' as a string with the number 30, which may lead to unexpected results due to type coercion. It would be safer to first convert 'year' to a number before performing the comparison.